File: app/lib/features/today/today_screen.dart
import 'package:flutter/material.dart';
import 'package:gym_coach/data/repositories/day_task_repository.dart';
import 'package:gym_coach/data/repositories/gamification_repository.dart';
import 'package:gym_coach/data/repositories/plan_repository.dart';
import 'package:gym_coach/features/today/today_controller.dart';
import 'package:gym_coach/theme/buttons.dart';
import 'package:gym_coach/theme/colors.dart';
import 'package:gym_coach/theme/mascot.dart';

class TodayScreen extends StatefulWidget {
  final PlanRepository planRepo;
  final DayTaskRepository taskRepo;
  final GamificationRepository gamifRepo;
  const TodayScreen({
    super.key,
    required this.planRepo,
    required this.taskRepo,
    required this.gamifRepo,
  });

  @override
  State<TodayScreen> createState() => _TodayScreenState();
}

class _TodayScreenState extends State<TodayScreen> {
  late final TodayController _ctrl = TodayController(
    planRepo: widget.planRepo,
    taskRepo: widget.taskRepo,
    gamifRepo: widget.gamifRepo,
  );

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.duoBackground,
      appBar: AppBar(title: const Text('今日任务'), backgroundColor: AppColors.duoGreen),
      body: FutureBuilder(
        future: _ctrl.load(),
        builder: (context, snap) {
          if (snap.connectionState != ConnectionState.done) {
            return const Center(child: CircularProgressIndicator());
          }
          final task = snap.data;
          if (task == null) {
            return const Center(child: Text('今天没有任务，去跟教练聊聊'));
          }
          return SingleChildScrollView(
            padding: const EdgeInsets.all(16),
            child: Column(
              children: [
                const AppMascot(mood: 'cheer'),
                const SizedBox(height: 16),
                if (task.workout != null)
                  Card(
                    child: ListTile(
                      title: Text(task.workout!.title),
                      subtitle: Text('${task.workout!.estimatedMinutes} 分钟 · ${task.workout!.exercises.length} 个动作'),
                      trailing: task.completedWorkout
                          ? const Icon(Icons.check_circle, color: AppColors.duoGreen)
                          : AppButton(
                              label: '完成',
                              onPressed: () async {
                                final xp = await _ctrl.completeWorkout(task);
                                if (mounted) {
                                  ScaffoldMessenger.of(context).showSnackBar(
                                    SnackBar(content: Text('+$xp XP')),
                                  );
                                }
                              },
                            ),
                    ),
                  ),
                ...task.meals.map((m) => Card(
                      child: ListTile(
                        title: Text('${m.slot} · ${m.name}'),
                        subtitle: Text('${m.kcal} kcal'),
                        trailing: task.completedMeals[m.slot] == true
                            ? const Icon(Icons.check_circle, color: AppColors.duoGreen)
                            : AppButton(
                                label: '吃了',
                                onPressed: () async {
                                  final xp = await _ctrl.completeMeal(task, m.slot);
                                  if (mounted) {
                                    ScaffoldMessenger.of(context).showSnackBar(
                                      SnackBar(content: Text('+$xp XP')),
                                    );
                                  }
                                },
                              ),
                      ),
                    )),
              ],
            ),
          );
        },
      ),
    );
  }
}

File: app/android/app/src/main/java/com/gymcoach/gym_coach/feature/today/TodayScreen.kt
package com.gymcoach.gym_coach.feature.today

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material3.Card
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.SnackbarHost
import androidx.compose.material3.SnackbarHostState
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import com.gymcoach.gym_coach.domain.model.DayTask
import com.gymcoach.gym_coach.theme.AppButton
import com.gymcoach.gym_coach.theme.AppMascot
import com.gymcoach.gym_coach.theme.AppColors
import com.gymcoach.gym_coach.theme.MascotMood
import kotlinx.coroutines.launch

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun TodayScreen(vm: TodayViewModel = hiltViewModel()) {
    val state by vm.state.collectAsState()
    val snackbar = remember { SnackbarHostState() }
    val scope = rememberCoroutineScope()
    LaunchedEffect(Unit) { vm.load() }
    Scaffold(
        snackbarHost = { SnackbarHost(snackbar) },
        containerColor = AppColors.duoBackground,
        topBar = {
            TopAppBar(
                title = { Text("今日任务") },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = AppColors.duoGreen,
                    titleContentColor = Color.White
                )
            )
        }
    ) { padding ->
        Box(
            modifier = Modifier.fillMaxSize().padding(padding),
            contentAlignment = Alignment.TopCenter
        ) {
            when (val s = state) {
                TodayUiState.Loading -> Text("加载中…")
                TodayUiState.Empty -> Text("今天没有任务，去跟教练聊聊")
                is TodayUiState.Loaded -> TodayContent(
                    task = s.task,
                    onCompleteWorkout = { vm.completeWorkout(s.task) { scope.launch { snackbar.showSnackbar("+$it XP") } } },
                    onCompleteMeal = { slot -> vm.completeMeal(s.task, slot) { scope.launch { snackbar.showSnackbar("+$it XP") } } }
                )
            }
        }
    }
}

@Composable
private fun TodayContent(
    task: DayTask,
    onCompleteWorkout: () -> Unit,
    onCompleteMeal: (String) -> Unit
) {
    Column(
        modifier = Modifier.fillMaxWidth().padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        AppMascot(mood = MascotMood.Cheer)
        task.workout?.let { w ->
            Card(modifier = Modifier.fillMaxWidth()) {
                Row(
                    modifier = Modifier.fillMaxWidth().padding(12.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Column(Modifier.weight(1f)) {
                        Text(w.title, style = MaterialTheme.typography.titleLarge)
                        Text("${w.estimatedMinutes} 分钟 · ${w.exercises.size} 个动作")
                    }
                    if (task.completedWorkout) {
                        Icon(Icons.Filled.CheckCircle, contentDescription = "done", tint = AppColors.duoGreen)
                    } else {
                        AppButton(label = "完成", onClick = onCompleteWorkout)
                    }
                }
            }
        }
        task.meals.forEach { meal ->
            Card(modifier = Modifier.fillMaxWidth()) {
                Row(
                    modifier = Modifier.fillMaxWidth().padding(12.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Column(Modifier.weight(1f)) {
                        Text("${meal.slot} · ${meal.name}", style = MaterialTheme.typography.titleLarge)
                        Text("${meal.kcal} kcal")
                    }
                    if (task.completedMeals[meal.slot] == true) {
                        Icon(Icons.Filled.CheckCircle, contentDescription = "done", tint = AppColors.duoGreen)
                    } else {
                        AppButton(label = "吃了", onClick = { onCompleteMeal(meal.slot) })
                    }
                }
            }
        }
    }
}
